### PR TITLE
(doc): Updated documentation for batch message send

### PR DESCRIPTION
**Pull request by vinagesh**
## Checklist
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-iot-sdk-csharp/blob/master/.github/CONTRIBUTING.md).
- [ ] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- [ ] This pull-request is submitted against the `master` branch.

## Description of the changes
 A customer was confused and had a question on the batch send. I have updated the comments to help clarify that.

### CODE REVIEW

**Review by drwill-ms — APPROVED** [review]

File: iothub/device/src/DeviceClient.cs [hunk 1]
```diff
@@ -482,14 +482,16 @@ public void SetRetryPolicy(IRetryPolicy retryPolicy)
         public Task SendEventAsync(Message message, CancellationToken cancellationToken) => InternalClient.SendEventAsync(message, cancellationToken);
 
         /// <summary>
-        /// Sends a batch of events to a hub. Requires AMQP or AMQP over WebSockets.
+        /// Sends a batch of events to IoT hub. Use AMQP or HTTPs for a true batch operation. MQTT will just send the messages one after the other.
+        /// For more information on routing <see href="https://docs.microsoft.com/en-us/azure/iot-edge/module-composition?view=iotedge-2018-06#declare-routes"/>
```
Comment (abhipsaMisra) [inline iothub/device/src/DeviceClient.cs:486]: this routing info wouldn't be relevant for the device client, correct?

File: iothub/device/src/ModuleClient.cs [hunk 2]
```diff
@@ -361,14 +361,16 @@ public void SetRetryPolicy(IRetryPolicy retryPolicy)
         public Task SendEventAsync(Message message, CancellationToken cancellationToken) => InternalClient.SendEventAsync(message, cancellationToken);
 
         /// <summary>
-        /// Sends a batch of events to IoT hub. Requires AMQP or AMQP over WebSockets.
+        /// Sends a batch of events to IoT hub. Use AMQP or HTTPs for a true batch operation. MQTT will just send the messages one after the other.
+        /// For more information on routing <see href="https://docs.microsoft.com/en-us/azure/iot-edge/module-composition?view=iotedge-2018-06#declare-routes"/>
```
Comment (abhipsaMisra) [inline iothub/device/src/ModuleClient.cs:365]: nit: maybe add a note that this routing info is only relevant for Edge modules (and not IoT modules)?

Comment (vinagesh) [reply iothub/device/src/ModuleClient.cs:365]: yes that makes sense.

File: iothub/device/src/DeviceClient.cs (hunk 1 above)
Comment (vinagesh) [reply iothub/device/src/DeviceClient.cs:486]: Yes good catch. Thanks.

**Pull request merged**